src/wmtk/simplex/link_condition.cpp

```diff
@@ -98,4 +98,93 @@ bool link_condition(const TriMesh& mesh, const Tuple& edge)
     return true;
 }
 
+bool link_condition_closed_tetmesh(const TetMesh& mesh, const Tuple& edge)
+{
+    // for closed mesh, if link(a) \intersect link(b) == link(ab)
+    const Simplex v_a = Simplex::vertex(edge);
+    const Simplex v_b = Simplex::vertex(mesh.switch_tuple(edge, PrimitiveType::Vertex));
+    const Simplex e_ab = Simplex::edge(edge);
+    const SimplexCollection link_a = link(mesh, v_a); // link(a)
+    const SimplexCollection link_b = link(mesh, v_b); // link(b)
+    const SimplexCollection link_ab = link(mesh, e_ab); // link(ab)
+
+    const SimplexCollection link_a_link_b_intersection =
+        SimplexCollection::get_intersection(link_a, link_b);
+
+    return SimplexCollection::are_simplex_collections_equal(link_a_link_b_intersection, link_ab);
+}
+
+bool link_condition(const TetMesh& mesh, const Tuple& edge)
+{
+    // close mesh check
+    if (!link_condition_closed_tetmesh(mesh, edge)) {
+        return false;
+    }
+
+    // add dummy vertex and link it to boundary faces
+    // the check can be convert from link_dummy(a) \intersect link_dummy(b) == link_dummy(ab) to:
+    // if ab is a boundary edge, then if any boundary face incident a shares a common edge with any
+    // boundary face incident b, return false.
+    // if ab is not boundary edge, then if a is not on
+    // boundary or b is not on boundary, return true, otherwise false
+
+    auto get_boundary_faces = [&mesh](const Tuple& _v) {
+        const Simplex input_v(PrimitiveType::Vertex, _v);
+        std::vector<Tuple> ret;
+        // get incident_faces from open_star
+        auto incident_faces = open_star(mesh, input_v).simplex_vector(PrimitiveType::Face);
+        for (const Simplex& _f : incident_faces) {
+            if (mesh.is_boundary(_f.tuple(), PrimitiveType::Face)) {
+                if (utils::SimplexComparisons::equal(
+                        mesh,
+                        Simplex(PrimitiveType::Vertex, _f.tuple()),
+                        input_v)) {
+                    // tuple edge is _v - x
+                    ret.push_back(mesh.switch_edge(mesh.switch_vertex(_f.tuple())));
+                } else {
+                    if ((utils::SimplexComparisons::equal(
```

Comment:
you can write that whole stuff as
```
if() {
} else if() {
} else {
}
```

For whatever reason you nested the second if-statement inside the else case. That gives you an unnecessary level of indention.

Additionally, this whole if-else stuff should not be necessary if you use `cofaces_single_dimension` because I think it guarantees that the tuple in there contains the input simplex. But check the documentation once more to make sure what I am saying is correct. And you can at least add an assert to your code here to be certain.